src/olympia/amo/management/commands/compress_assets.py

```diff
@@ -0,0 +1,302 @@
+
+    def generate_build_id(self, use_uuid):
+        if use_uuid:
+            return uuid.uuid4().hex[:8]
+        else:
+            root = getattr(settings, 'JINGO_MINIFY_ASSETS_GIT_ROOT', '.')
+            git_bin = getattr(settings, 'GIT_BIN', 'git')
+            return check_output(
+                [git_bin, '-C', root, 'rev-parse', '--short', 'HEAD']).strip()
+
+    def update_hashes(self, update=False):
+        if update:
+            # Adds a time based hash on to the build id.
+            self.build_id = '%s-%s' % (
+                self.build_id, hex(int(time.time()))[2:])
+
+        build_id_file = os.path.realpath(os.path.join(settings.ROOT,
+                                                      'build.py'))
+        with open(build_id_file, 'w') as f:
+            f.write('BUILD_ID_CSS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_JS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_IMG = "%s"\n' % self.build_id)
+            f.write('BUNDLE_HASHES = %s\n' % self.bundle_hashes)
+
+    def handle(self, **options):
+        self.build_id = self.generate_build_id(options.get('use_uuid', False))
+
+        if options.get('do_update_only', False):
+            self.update_hashes(update=True)
+            return
+
+        jar_path = (os.path.dirname(__file__), '..', '..', 'bin',
+                    'yuicompressor-2.4.7.jar')
+        self.path_to_jar = os.path.realpath(os.path.join(*jar_path))
+
+        self.v = '-v' if options.get('verbosity', False) == '2' else ''
+
+        cachebust_imgs = getattr(settings, 'CACHEBUST_IMGS', False)
+        if not cachebust_imgs:
+            print 'To turn on cache busting, use settings.CACHEBUST_IMGS'
+
+        # This will loop through every bundle, and do the following:
+        # - Concat all files into one
+        # - Cache bust all images in CSS files
+        # - Minify the concatted files
+
+        for ftype, bundle in settings.MINIFY_BUNDLES.iteritems():
+            for name, files in bundle.iteritems():
+                # Set the paths to the files.
+                concatted_file = path(ftype, '%s-all.%s' % (name, ftype,))
+                compressed_file = path(ftype, '%s-min.%s' % (name, ftype,))
+
+                files_all = []
+                for fn in files:
+                    processed = self._preprocess_file(fn)
+                    # If the file can't be processed, we skip it.
+                    if processed is not None:
+                        files_all.append(processed)
+
+                # Concat all the files.
+                tmp_concatted = '%s.tmp' % concatted_file
+                if len(files_all) == 0:
+                    raise CommandError(
+                        'No input files specified in '
+                        'MINIFY_BUNDLES["%s"]["%s"] in settings.py!' %
+                        (ftype, name)
+                    )
+                self._call(
+                    'cat %s > %s' % (' '.join(files_all), tmp_concatted),
```

Comment:
This I knew already and... well, this is terrible.

Reply:
:see_no_evil: it is indeed, crazily terrible. I didn't dare to touch any of that.